### PR TITLE
Weather app - Cornelia 

### PR DESCRIPTION
**Pull request by Corneliaryas**
Link to my deployed project:  https://weather-project-cornelia.netlify.app/

My reflections on how this week's project turned out: I really enjoyed this project and I'm happy with the result. Feels great to see an example of what we could use our new skills for. 

Things I'd like to have clarified or explained in more detail: How I can optimize my code better, especially the CSS in this case (but also the JS and HTML of course). I also feel like the geolocation could have been done in a simpler way maybe?

### CODE REVIEW

File: code/style.css [hunk 1]
```diff
@@ -0,0 +1,665 @@
...
+    .current-main {
+        width: 60%;
+    }
+
+    .forecast {
+        width: 60%;
+    }
+
+}
+
```
Comment (JHarde) [inline code/style.css:665]: Your CSS code looks pretty flawless to me, could you have done anything to have fewer lines of code in your CSS? It really looks like you are mastering flex, perhaps try to challenge yourself by trying out grid in the next project? 🎖️

**Review by JHarde — COMMENTED** [review]
Your website looks super nice!! A lot of functionality, clean code and great animations. I can tell that you were striving towards black level and you most definitely achieved it. You should be really proud of your project and the work you have done! 🏅 👍

**Review by JHarde — APPROVED** [review]

File: code/index.html [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            <div></div>
+        </div>
+        <div class="desktop-header">
+            <h2>Weather</h2>
+            <nav id="nav-bar">
+                <a href="#" onclick="showCurrentWeather(stockholmCurrentAPI), showWeatherForecast(stockholmOneCallAPI);">Stockholm</a>
+                <a href="#" onclick="showCurrentWeather(umeaCurrentAPI), showWeatherForecast(umeaOneCallAPI);">Umeå</a>
+                <a href="#" onclick="showCurrentWeather(klintanCurrentAPI), showWeatherForecast(klintanOneCallAPI);">Robertsfors</a>
+                <a href="#" onclick="showCurrentWeather(bodenCurrentAPI), showWeatherForecast(bodenOneCallAPI);">Boden</a>
+                <a href="#" onclick="getLocationAPI(setLocationAPI);" class="din-position"><img src="./assets/GeoIcon.svg"/>Din position</a>
```
Comment (HIPPIEKICK) [inline code/index.html:24]: Maybe buttons would make more sense than a-tags? ;)

File: code/script.js [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+const bodenCurrentAPI = 'https://api.openweathermap.org/data/2.5/weather?q=Boden,Sweden&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723&lang=sv';
+const klintanCurrentAPI = 'https://api.openweathermap.org/data/2.5/weather?q=Robertsfors,Sweden&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723&lang=sv';
+const stockholmOneCallAPI = 'https://api.openweathermap.org/data/2.5/onecall?lat=59.3293&lon=18.0686&exclude=current,minutely,hourly,alerts&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723'
+const umeaOneCallAPI = 'https://api.openweathermap.org/data/2.5/onecall?lat=63.8258&lon=20.2630&exclude=current,minutely,hourly,alerts&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723'
+const bodenOneCallAPI = 'https://api.openweathermap.org/data/2.5/onecall?lat=65.8251&lon=21.6887&exclude=current,minutely,hourly,alerts&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723'
+const klintanOneCallAPI = 'https://api.openweathermap.org/data/2.5/onecall?lat=64.1918&lon=20.8489&exclude=current,minutely,hourly,alerts&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723'
+
```
[the remark this answers is not in the document]
Comment (HIPPIEKICK) [reply code/script.js:10]: Agreed! And not sure what OneCall means 🤔

File: code/script.js [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+             navigator.geolocation.getCurrentPosition(setCoordinates, handleError); 
+         } else {
+            showDefaultCity();
+         }
+    }
+    const setCoordinates = (position) => {
+         lat = position.coords.latitude
+         lon = position.coords.longitude
+         geoLocationAPI = `https://api.openweathermap.org/data/2.5/weather?lat=${lat}&lon=${lon}&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723&lang=sv`
+         geoLocationOneCallAPI = `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${lon}&exclude=current,minutely,hourly,alerts&units=metric&appid=2163e0bcc8eaa7f0951284d8a650a723`
```
Comment (HIPPIEKICK) [inline code/script.js:52]: These could also be variables on top of the file

File: code/script.js [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+    showWeatherForecast(geoLocationOneCallAPI);
+}
+
+//Shows or hides the navigation bar
+
+const showHamburgerMenu = () => {
+    if(navBar.style.display === "none") {
+        navBar.style.display = "flex";
+        hamburgerMenu.style.flexDirection = "row"
+
```
[the remark this answers is not in the document]
Comment (HIPPIEKICK) [reply code/script.js:89]: Try to keep the styling to the CSS-file and instead toggle CSS classes in the JS-file

File: code/script.js [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+        hamburgerMenu.style.flexDirection = "column"
+    }
+}
+
+//Shows or hides the weather animations
+
+const showCloudAnimation = () => {
+    hideWeatherAnimation();
+    cloudAnimation.style.display = "block";
+    currentWeatherMain.classList.add('clouds');
```
Comment (HIPPIEKICK) [inline code/script.js:111]: That's the way ⭐

File: code/script.js [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+                    currentWeatherMain.classList.add('snow');
+                    currentWeatherMain.classList.remove('clouds');
+                } else {
+                    hideWeatherAnimation();
+                }
+            }
+            setWeatherColors();
+            currentWeatherContainer.innerHTML = `<h1>${weatherTemp}<sup>&degC</sup></h1><h2>${weather.name}</h2><p>${weatherDescription}</p>`
+            sunriseSpan.innerHTML = `<p> ${sunriseString} </p>`
+            sunsetSpan.innerHTML = `<p> ${sunsetString} </p>`
```
Comment (HIPPIEKICK) [inline code/script.js:177]: Do you need the p-tags here, or can you just insert this into the span:
```
sunriseSpan.innerHTML = sunriseString
sunsetSpan.innerHTML = sunsetString
```

File: code/script.js [hunk 8]
```diff
@@ -0,0 +1,219 @@
...
+                forecastHTML += ` </div>`;
+                forecastHTML += `</div>`;
+            
+                weatherForecastContainer.innerHTML += `${forecastHTML}`;    
+            }))
+
+            minMax.innerHTML = `<p>Min ${todayMin}&degC / Max ${todayMax}&degC</p>`
+    
+        })
+    }
```
Comment (HIPPIEKICK) [inline code/script.js:214]: Indentation ;)

**Review by HIPPIEKICK — APPROVED** [review]
Great job Cornelia! Useful comments and your code is easy to follow. Keep it up 💯

**Pull request closed without merging**